src/databricks/labs/lakebridge/transpiler/installers.py

```diff
@@ -549,3 +564,252 @@ def _parse_java_version(cls, version: str) -> tuple[int, int, int, int] | None:
         update = int(match["update"] or 0)
         patch = int(match["patch"] or 0)
         return feature, interim, update, patch
+
+
+class SwitchInstaller(TranspilerInstaller):
+    _INSTALL_STATE_KEY = "Switch"
+    _TRANSPILER_ID = "switch"
+    _PYPI_PACKAGE_NAME = "databricks-switch-plugin"
+
+    @property
+    def name(self) -> str:
+        return self._INSTALL_STATE_KEY
+
+    def can_install(self, artifact: Path) -> bool:
+        wheel_name = self._PYPI_PACKAGE_NAME.replace("-", "_")
+        return wheel_name in artifact.name and artifact.suffix == ".whl"
+
+    def install(self, artifact: Path | None = None) -> bool:
```

Comment:
This is how we have separated things out: the installer installs things locally, deployer deploys to the workspace. You need something similar to recon deployer that way you don't need to have workspace client inside `TranspilerInstaller`